microprofile.ls/org.eclipse.lsp4mp.ls/src/main/java/org/eclipse/lsp4mp/services/MicroProfileDefinition.java

```diff
@@ -180,20 +231,48 @@ private static MicroProfilePropertyDefinitionParams getPropertyDefinitionParams(
 		return definitionParams;
 	}
 
+	private static Either<List<? extends Location>, List<? extends LocationLink>> getPropertyDefinition(
+			PropertiesModel document, boolean definitionLinkSupport, PropertyValueExpression propertyValueExpression,
+			List<Property> referencedProperties) {
+
+		if (definitionLinkSupport) {

```

Comment:
To avoid having 2 code for managing Location and LocationLink, it should be better to manage only LocationLink. The ApplicationTextDocumentService consumes the MicroProfileDefinition which returns LocationLink only and convert to Location if !definitionLinkSupport. 

We does that in LemMinX, see https://github.com/eclipse/lemminx/blob/d2d5bb76a1309f6d52bd8177d08aecbfafa5b5fb/org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/XMLTextDocumentService.java#L335